includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -151,4 +151,58 @@ protected function unwrap_p_element( DOMElement $node ) {
 			$parent_node->parentNode->replaceChild( $node, $parent_node );
 		}
 	}
+
+	/**
+	 * Removes the node's nearest `<script>` sibling with a `src` attribute containing the base `src` URL provided.
+	 *
+	 * @since 2.1
+	 *
+	 * @param DOMElement $node         The DOMNode to whose sibling is the script to be removed.
+	 * @param string     $base_src_url Script URL to match against.
+	 * @param string     $content      Text content of node to match against.
+	 */
+	protected function maybe_remove_script_sibling( DOMElement $node, $base_src_url, $content = '' ) {
+		$next_element_sibling = $node->nextSibling;
+
+		while ( $next_element_sibling && ! ( $next_element_sibling instanceof DOMElement ) ) {
+			$next_element_sibling = $next_element_sibling->nextSibling;
+		}
+
+		// Handle case where script is wrapped in paragraph by wpautop.
+		if ( $next_element_sibling instanceof DOMElement && 'p' === $next_element_sibling->nodeName ) {
```

Comment:
Based on a suggestion from @schlessera I've started using `tagName` instead of `nodeName` when we know the node is an element.

```suggestion
		if ( $next_element_sibling instanceof DOMElement && 'p' === $next_element_sibling->tagName ) {
```

Could be done for the other instances here too.